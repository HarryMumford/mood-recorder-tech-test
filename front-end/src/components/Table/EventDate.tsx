import * as React from 'react';
import { getDay, getDate, getMonth } from '../../services/dateFormatter';
import styled from 'styled-components';

const FullDate = styled.div`
  font-size: 0.7rem;
  color: darkslategray;
`;

const EventDate = (props: { timestamp: string }) => {
  const { timestamp } = props;
  const day = getDay(timestamp);
  const date = getDate(timestamp);
  const month = getMonth(timestamp);
  const year = new Date(timestamp).getFullYear();

  return (
    <>
      <FullDate key={timestamp}>{`${day} ${date} ${month} ${year}`}</FullDate>
    </>
  );
};

export default EventDate;
