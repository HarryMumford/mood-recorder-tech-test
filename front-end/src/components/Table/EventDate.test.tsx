import * as React from 'react';
import { shallow } from 'enzyme';
import EventDate from './EventDate';

describe('EventDate', () => {
  it('should render correctly with given timestamp', () => {
    const strings = '1598457735';
    const component = shallow(<EventDate timestamp={strings} />);
    console.log(component);
    expect(component).toMatchSnapshot();
  });
});
