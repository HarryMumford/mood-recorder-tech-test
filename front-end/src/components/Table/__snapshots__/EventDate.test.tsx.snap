// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDate should render correctly with given timestamp 1`] = `
<styled.td>
  undefined NaNth undefined NaN
</styled.td>
`;

exports[`Table container should render banner text correctly with given strings 1`] = `
<styled.td>
  undefined NaNth undefined NaN
</styled.td>
`;

exports[`Table container should render correctly with given timestamp 1`] = `
<styled.td>
  undefined NaNth undefined NaN
</styled.td>
`;

exports[`Table container should render the container correctly 1`] = `
<styled.div>
  <Rows />
</styled.div>
`;
